File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while True:
        h0 = heapq.heappop(scoville)
        if h0 >= K:
            break
        if len(scoville)<1:
            return -1
        h1 = heapq.heappop(scoville)
        heapq.heappush(scoville, h0+(h1*2))

        answer+=1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> hp = new PriorityQueue<>();
        for(int s: scoville){
            hp.offer(s);
        }
        
        while(hp.peek() < K){
            answer++;
            if(hp.size() < 2) return -1;
            int s1 = hp.poll();
            int s2 = hp.poll();
            int mixed = s1+(s2*2);
            hp.offer(mixed);
        }
        return answer;
    }
}